pkg/api/bzz_test.go

```diff
@@ -0,0 +1,143 @@
+)
+
+func TestBzz(t *testing.T) {
+	var (
+		bzzDownloadResource = func(addr, path string) string { return "/bzz:/" + addr + "/" + path }
+		storer              = smock.NewStorer()
+		sp                  = splitter.NewSimpleSplitter(storer)
+		client              = newTestServer(t, testServerOptions{
+			Storer:         storer,
+			ManifestParser: jsonmanifest.NewParser(),
+			Tags:           tags.NewTags(),
+			Logger:         logging.New(ioutil.Discard, 5),
+		})
+	)
+
+	t.Run("download-file-by-path", func(t *testing.T) {
+		fileName := "sample.html"
+		filePath := "test/" + fileName
+		missingFilePath := "test/missing"
+		sampleHtml := `<!DOCTYPE html>
+		<html>
+		<body>
+	
+		<h1>My First Heading</h1>
+	
+		<p>My first paragraph.</p>
+	
+		</body>
+		</html>`
+
+		var err error
+		var fileReference swarm.Address
+		var manifestFileReference swarm.Address
+
+		// save file
+
+		fileReference, err = file.SplitWriteAll(context.Background(), sp, strings.NewReader(sampleHtml), int64(len(sampleHtml)), false)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// save manifest
+
+		jsonManifest := jsonmanifest.NewManifest()
+
+		jsonManifest.Add(filePath, jsonmanifest.JSONEntry{
+			Reference: fileReference,
+			Name:      fileName,
+			Headers: http.Header{
+				"Content-Type": {"text/html", "charset=utf-8"},
+			},
+		})
+
+		manifestFileBytes, err := jsonManifest.Serialize()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		fr, err := file.SplitWriteAll(context.Background(), sp, bytes.NewReader(manifestFileBytes), int64(len(manifestFileBytes)), false)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		m := entry.NewMetadata(fileName)
+		m.MimeType = api.ManifestContentType
+		metadataBytes, err := json.Marshal(m)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		mr, err := file.SplitWriteAll(context.Background(), sp, bytes.NewReader(metadataBytes), int64(len(metadataBytes)), false)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// now join both references (mr,fr) to create an entry and store it.
+		newEntry := entry.New(fr, mr)
+		fileEntryBytes, err := newEntry.MarshalBinary()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		manifestFileReference, err = file.SplitWriteAll(context.Background(), sp, bytes.NewReader(fileEntryBytes), int64(len(fileEntryBytes)), false)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// read file from manifest path
+
+		rcvdHeader := jsonhttptest.ResponseDirectCheckBinaryResponse(t, client, http.MethodGet, bzzDownloadResource(manifestFileReference.String(), filePath), nil, http.StatusOK, []byte(sampleHtml), nil)
+		cd := rcvdHeader.Get("Content-Disposition")
+		_, params, err := mime.ParseMediaType(cd)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if params["filename"] != fileName {
+			t.Fatal("Invalid file name detected")
+		}
+		if rcvdHeader.Get("ETag") != fmt.Sprintf("%q", fileReference) {
+			t.Fatal("Invalid ETags header received")
+		}
+		if rcvdHeader.Get("Content-Type") != "text/html; charset=utf-8" {
```

Comment:
can we extract `text/hml; charset=utf-8` as a `const`?

Reply:
This was left as is, because it was used as here on 3 places in `file_test.go` file, and I figured there was a reason for it.
It I were to extract it, it would be changed for that other file.

Since this is for test, and there is no other point for this string, I would leave it as is. It was meant to illustrate point from [pkg/api/bzz_test.go#78](https://github.com/ethersphere/bee/pull/452/files/5f94d6c9d4306315ca4ce316d5efd9f14596bcc0#diff-804afbcbcdabfa91b45c34d0af02358eR78).
If we ever need some greater dealings for this to be `const`, it would probably be better to have another package that contains commonly used mime types.

Reply:
fair enough